dart/constraint/ConstraintSolver.cpp

```diff
@@ -436,16 +436,19 @@ DART_SUPPRESS_DEPRECATED_BEGIN
     shapeFrame2->asShapeNode()->getBodyNodePtr()->setColliding(true);
 DART_SUPPRESS_DEPRECATED_END
 
+    // If penetration depth is negative, then the collision isn't really
+    // happening and the contact point should be ignored.
+    // TODO(MXG): Investigate ways to leverage the proximity information of a
+    //            negative penetration to improve collision handling.
+    if (contact.penetrationDepth >= 0.0)
```

Comment:
Oh, wait. Actually, this should be:
```cpp
if (contact.penetrationDepth < 0.0)
```
to match the original logic.

Reply:
Whoops! Thank goodness for unit tests :sweat_smile: